docs/docsite/rst/user_guide/complex_data_manipulation.rst

```diff
@@ -0,0 +1,71 @@
+Complex data manipulation
+#########################
+
+In many cases you need to do some complex operation with your variables, while Ansible is not recommended as a data processing/manipulation tool, you can use the existing Jinja2 templating in conjunction with the many added ansible filters, lookups and tests to do some very complex transformations.
+
+A quick definition of each type of plugin before we go forward
+  - lookups: Mainly used to query 'external data', in Ansible these were the primary part of loops using the `with_<lookup>` construct, but they can be used independantly to return data for processing. They normally returna list due to their primary function in loops as mentioned previouslly. Used with the ``lookup`` or ``query`` Jinja2 operators.
+  - filters: used to change/transform data, used mainly with the ``|`` Jinja2 operator.
+  - tests: used to validate data, used mainly with the ``is`` Jinja2 operator.
+
+
+.. _for_loops_or_list_comprehensions:
+For loops and/or list comprehensions
+====================================
+
+Most programmers are used to using a for loop (or while and others) and list comprehensions to do transformations on lists or lists of objects, Jinja2 has a few filters that provide this functionality: map, select, reject, selectattr, rejectattr
+
+ - map: this is a basic for loop that just allows you to transform every item in a list, using the 'attribute' keyword you can do the transformation based on a attribute of the list element.
+
```

Comment:
May need to list the Jinja2 minimum version here? (Or do we not have to worry about that because we're no longer supporting installing on the RHEL6 python-2.6 stack?)

Reply:
added note